protocols/gossipsub/src/transform.rs

```diff
@@ -0,0 +1,58 @@
+//! This trait allows of extended user-level decoding that can apply to message-data before a
```

Comment:
Can you add a license header here?

Reply:
Can do. 

I think its mostly stable now. Will fix the doc links.